ext/php5/dispatch.c

```diff
@@ -21,7 +21,7 @@ extern inline void ddtrace_dispatch_release(ddtrace_dispatch_t *dispatch);
 
 static bool dd_try_find_function_dispatch(HashTable *ht, zval *fname, ddtrace_dispatch_t **dispatch_ptr,
                                           HashTable **function_table) {
-    ddtrace_dispatch_t *dispatch = ddtrace_hash_find_ptr_lc(ht, Z_STRVAL_P(fname), Z_STRLEN_P(fname));
+    ddtrace_dispatch_t *dispatch = ddtrace_hash_find_ptr_lc(ht, Z_STRVAL_P(fname), Z_STRLEN_P(fname) + 1);
```

Comment:
This looks like it is allocating one extra byte for the [`zend_str_tolower_dup` call](https://github.com/DataDog/dd-trace-php/blob/d3bc9f01302a94bc94465507fc61a01968da050d/ext/php5/arrays.c#L32). What do you think about adding the `+ 1` to the [`zend_hash_find` call from within `ddtrace_hash_find_ptr`](https://github.com/DataDog/dd-trace-php/blob/d3bc9f01302a94bc94465507fc61a01968da050d/ext/php5/arrays.c#L19) to avoid copying PHP 5-isms into the ddtrace API?

Reply:
I think that's going to be more confusing than not, especially given that we do not use only ddtrace_hash_ functions, but also zend_hash_. (like, the one call is using +1, but the other isn't, wtf?)

Reply:
Regarding the extra byte, should the code in `ddtrace_hash_find_ptr` do a -1 then?

Reply:
If this were a ZAI API, I would say for sure don't propagate PHP 5-isms to the API. But in this case I'm 👍 with adding a `- 1` to `zend_str_tolower_dup` if you find that preferable.